src/mlpack/methods/ne/genome.hpp

```diff
@@ -0,0 +1,334 @@
+/**
+ * @file genome.hpp
+ * @author Bang Liu
+ *
+ * Definition of the Genome class.
+ */
+#ifndef MLPACK_METHODS_NE_GENOME_HPP
+#define MLPACK_METHODS_NE_GENOME_HPP
+
+#include <cstddef>
+#include <cassert>
+#include <map>
+
+#include <mlpack/core.hpp>
+
+#include "link_gene.hpp"
+#include "neuron_gene.hpp"
+#include "utils.hpp"
+
+namespace mlpack {
+namespace ne {
+
+/**
+ * This class defines a genome.
+ # A genome is consist of a group of genes.
+ */
+class Genome {
+ public:
+  // Neurons.
+  std::vector<NeuronGene> aNeuronGenes;
+
+  // Links.
+  std::vector<LinkGene> aLinkGenes;
+
+  // Default constructor.
+  Genome() {
+    aId = -1;
+    aNumInput = 0;
+    aNumOutput = 0;
+    aFitness = DBL_MAX;
+  }
+  
+  // Parametric constructor.
+  Genome(int id,
+  	     const std::vector<NeuronGene>& neuronGenes,
+         const std::vector<LinkGene>& linkGenes,
+         int numInput,
+         int numOutput,
+         double fitness):
+    aId(id),
+    aNeuronGenes(neuronGenes),
+    aLinkGenes(linkGenes),
+    aNumInput(numInput),
+    aNumOutput(numOutput),
+    aFitness(fitness)
+  {}
+
+  // Copy constructor.
+  Genome(const Genome& genome) {
+    aId = genome.aId;
+    aNeuronGenes = genome.aNeuronGenes;
+    aLinkGenes = genome.aLinkGenes;
+    aNumInput = genome.aNumInput;
+    aNumOutput = genome.aNumOutput;
+    aFitness = genome.aFitness;
+  }
+
+  // Destructor.
+  ~Genome() {}
+
+  // Operator =.
+  Genome& operator =(const Genome& genome) {
+    if (this != &genome) {
+      aId = genome.aId;
+      aNeuronGenes = genome.aNeuronGenes;
+      aLinkGenes = genome.aLinkGenes;
+      aNumInput = genome.aNumInput;
+      aNumOutput = genome.aNumOutput;
+      aFitness = genome.aFitness;
+    }
+
+    return *this;
+  }
+
+  // Get genome id.
+  int Id() const { return aId; }
+
+  // Set genome id.
+  void Id(int id) { aId = id; }
```

Comment:
Generally we use a setter function that takes the input as an argument if it's necessary to validate what the user is setting the value to. Otherwise, we use a function that allows a direct access e.g.:

```
//! Get the genome id.
int Id() const { return id; }
//! Modify the genome id.
int& Id() { return id; }
```